sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -454,7 +454,9 @@ case class InSet(child: Expression, hset: Set[Any]) extends UnaryExpression with
   }
 
   override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
-    if (canBeComputedUsingSwitch && hset.size <= SQLConf.get.optimizerInSetSwitchThreshold) {
+    val sqlConf = SQLConf.get
+    if (canBeComputedUsingSwitch && hset.size <= sqlConf.optimizerInSetSwitchThreshold &&
+      sqlConf.codegenUseSwitchStatement) {
```

Comment:
instead of an added configuration, could this be a `try / catch`?  first try with switch, if it fails, then log & error and use if-else?  avoids the users having to set a config, and then unset it again when we have the right fix from janino

Reply:
Yeah it would be ideal. Thanks! I over-thought a bit and thought it would be hard to deal with it, but looks like not that complicated. Let me update the code and PR information.